src/components/common/vrn-capture-modal/vrn-capture-modal.ts

```diff
@@ -0,0 +1,73 @@
+import { Component } from '@angular/core';
+import {
+  AbstractControl, FormControl, FormGroup, Validators,
+} from '@angular/forms';
+import { NavParams } from '@ionic/angular';
+import {
+  FieldValidators,
+  getRegistrationNumberValidator,
+  nonAlphaNumericValues,
+} from '@shared/constants/field-validators/field-validators';
+
+@Component({
+  selector: 'vrn-capture-modal',
+  templateUrl: './vrn-capture-modal.html',
+  styleUrls: ['./vrn-capture-modal.scss'],
+})
+export class VRNCaptureModal {
+
+  onCancel: Function;
+
+  onSave: Function;
+
+  vehicleRegistrationNumber: string;
+
+  formGroup: FormGroup;
+
+  vehicleRegistrationFormControlName: string = 'vehicleRegistration';
+
+  formInvalid: boolean = false;
+
+  readonly registrationNumberValidator: FieldValidators = getRegistrationNumberValidator();
+
+  constructor(
+    private navParams: NavParams,
+  ) {
+    this.onCancel = this.navParams.get('onCancel');
```

Comment:
not sure this is the recommended way of passing data between page and modal.  Think we should be using `dismiss()` and `onDidDismiss()` methods (see [docs](https://ionicframework.com/docs/api/modal#dismissing-a-modal)) without need for `navParams`